tests/cloud/conftest.py

```diff
@@ -0,0 +1,170 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c)  2023-2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# -----------------------------------------------------------------------------
+# "gpt2-medium","TinyLlama/TinyLlama-1.1B-Chat-v1.0","Salesforce/codegen-350M-mono","wtang06/mpt-125m-c4"
+import os
+import pytest
+import QEfficient
+import json
+from typing import List
+from QEfficient.utils.constants import Constants, QEFF_MODELS_DIR, ROOT_DIR
+
+class model_setup:
+    """
+    model_setup is a set up class for all the High Level testing script, 
+    which provides all neccessary objects needed for checking the flow and creation 
+    of the HL API code.
+    """
+    def __init__(self,model_name,num_cores,prompt,prompts_txt_file_path,aic_enable_depth_first,mos,cache_dir,hf_token,batch_size,prompt_len,ctx_len,mxfp6,mxint8,device_group):
+        """
+        Initialization set up
+        ------
+        param: model_name: str
+        param: num_cores: int
+        param: prompt: str
+        param: prompts_txt_file_path: str
+        param: aic_enable_depth_first: bool 
+        param: mos: int
+        param: cache_dir: str 
+        param: hf_token: str 
+        param: batch_size: int
+        param: prompt_len: int 
+        param: ctx_len: int 
+        param: mxfp6: bool 
+        param: mxint8: bool
+        param: device_group: List[int] 
+        """
+        self.model_name = model_name
+        self.num_cores = num_cores
+        self.prompt = prompt
+        self.prompts_txt_file_path = os.path.join(ROOT_DIR,prompts_txt_file_path) if prompts_txt_file_path is not None  else None
+        self.aic_enable_depth_first = aic_enable_depth_first
+        self.mos = mos
+        self.cache_dir = cache_dir
+        self.hf_token = hf_token
+        self.batch_size = batch_size
+        self.prompt_len = prompt_len
+        self.ctx_len = ctx_len
+        self.mxfp6 = mxfp6
+        self.mxint8 = mxint8
+        self.device_group = device_group
+
+    def model_card_dir(self):
+        return str(os.path.join(QEFF_MODELS_DIR, str(self.model_name)))
+    
+    def qpc_base_dir_name(self):

```

Comment:
use from QEfficient.utils

Reply:
This is just setup object which contains all the parameters for a particular test case, For each test case one setup object will be created and it is easy keep track of all the parameters.
There is not enough functionality present in the **QEfficient.utils**.
Yes, We can remove this setup class but there will be some redundant code.

Reply:
discussed.